contrib/macos/Start DOSBox Staging (logging).command

```diff
@@ -0,0 +1,22 @@
+#!/bin/bash
+
+CWD=$(dirname "$0")
+COMMAND=$(basename "$0")
+
+NUM_COLUMNS=120
+NUM_ROWS=45
+FONT_SIZE=14
+
+osascript <<END
+tell application "Terminal"
+  set number of columns of (every window whose name contains "$COMMAND") to $NUM_COLUMNS
+  set number of rows    of (every window whose name contains "$COMMAND") to $NUM_ROWS
+  set font size         of (every window whose name contains "$COMMAND") to $FONT_SIZE
+  set background color  of (every window whose name contains "$COMMAND") to {5000,5000,5000,0}
+  set normal text color of (every window whose name contains "$COMMAND") to {40000,40000,40000,0}
+end tell
+END
+
+/Applications/DOSBox\ Staging.app/Contents/MacOS/dosbox --working-dir "$CWD"
+
+(sleep 0.1 ; osascript -e "tell application \"Terminal\" to close (every window whose name contains \"$COMMAND\")") &
```

Comment:
Is it possible to somehow exit this shell script, but leave its child processes around as zombies? (like the background `(sleep 0.1 ; osascript) &`?

bash's `trap` command sets up an exit handler, like a C++ destructor, that triggers on an exit signal.

It's great for scripts that might create temp files, where the user might `Ctrl+C` right in the middle of some job - the `trap` can always cleanup, no matter where it exited.

In this case, I'm thinking we could using something like:

``` bash
trap 'kill $(jobs -p)' EXIT
```

Where `jobs -p` gets the script's child PIDs, so would mop up any errant DOSBox Staging and backgrounded `osascript` processes.

The `EXIT` will run on any exit, but it can be expanded with a list of SIGNALs (like: `SIGINT`, `SIGTERM`, etc..) or you can be very specific, like just `SIGINT` to catch `Ctrl+C` breaks.

Just something to try, if it helps (or there's a risk of dangling children).

Reply:
Great tips @kcgen; I've learned something new!

I've enhanced the scripts a bit, now the logging one handles Ctrl-C-ing the terminal window itself too, and they're both a bit more clever—they'll quite the Terminal app too if there are no other open windows.